File: tooling/cli/templates/plugin/android/src/main/ExamplePlugin.kt
package {{package_id}}

import app.tauri.plugin.JSObject
import app.tauri.plugin.Plugin
import app.tauri.plugin.Invoke
import app.tauri.plugin.PluginMethod

class ExamplePlugin: Plugin() {
    private val implementation = Example()

    @PluginMethod
    fun ping(invoke: Invoke) {
        val value = invoke.getString("value") ?: ""
        val ret = JSObject()
        ret.put("value", implementation.pong(value))
        invoke.resolve(ret)
    }
}

File: tooling/cli/templates/plugin/ios/Sources/ExamplePlugin.swift
import UIKit
import WebKit
import Tauri
import SwiftRs

class ExamplePlugin: Plugin {
	@objc public func ping(_ invoke: Invoke) throws {
		let value = invoke.getString("value")
		invoke.resolve(["value": value as Any])
	}
}

@_cdecl("init_plugin_{{ plugin_name_snake_case }}")
func initPlugin() -> Plugin {
	return ExamplePlugin()
}
